listener/listener.go

```diff
@@ -260,37 +286,32 @@ func (l *Listener) Listen(ctx context.Context, handler message.Handler, topicNam
 	}()
 
 	// Generate new subscription client
+	// TODO: Add servicebus.SubscriptionWithPrefetchCount(prefetch count) option and expose the setting via the listener
 	sub, err := topic.NewSubscription(l.subscriptionEntity.Name)
 	if err != nil {
 		return fmt.Errorf("failed to create new subscription %s: %w", l.subscriptionEntity.Name, err)
 	}
-	subReceiver, err := sub.NewReceiver(ctx)
+
+	var receiverOpts []servicebus.ReceiverOption
+	if l.prefetchCount != nil {
+		receiverOpts = append(receiverOpts, servicebus.ReceiverWithPrefetchCount(*l.prefetchCount))
+	}
+	handlerConcurrency := 1
+	if l.maxConcurrency != nil {
+		handlerConcurrency = *l.maxConcurrency
+	}
+	subReceiver, err := sub.NewReceiver(ctx, receiverOpts...)
 	if err != nil {
 		return fmt.Errorf("failed to create new subscription receiver %s: %w", l.subscriptionEntity.Name, err)
 	}
-
-	// Create a handle class that has that function
-	listenerHandle := subReceiver.Listen(ctx, servicebus.HandlerFunc(
-		func(ctx context.Context, msg *servicebus.Message) error {
-			ctx, s := tracing.StartSpanFromMessageAndContext(ctx, "go-shuttle.Listener.HandlerFunc", msg)
-			defer s.End()
-			if l.lockRenewalInterval != nil {
-				renewer := peeklock.RenewPeriodically(ctx, *l.lockRenewalInterval, sub, msg)
-				defer renewer.Stop()
-			}
-			currentHandler := handler
-			for !message.IsDone(currentHandler) {
-				currentHandler = currentHandler.Do(ctx, handler, msg)
-				// handle nil as a Completion!
-				if currentHandler == nil {
-					currentHandler = message.Complete()
-				}
-			}
-			return nil
-		}))
+	listenerHandle := subReceiver.Listen(ctx,
```

Comment:
Seems good as is. 

However, I am curious what making the handlers public is meant to achieve? Are we thinking the user of this lib, might construct their own subReceiver and their own set of handlers? Seems like a viable; although, unlikely option. Simply curious :)

Reply:
yes. as made clear in the description, these handler are not at all tied to go-shuttle. they can be used with the raw sdk, and they will likely move there in the future.